pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -980,14 +1073,45 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 			return a.deleteNode(ctx, node)
 		}
 
-		// node is deleted, we can power on the host
+		//we need this annotation to differentiate between unhealthy machine that
+		//needs remediation, and an unhealthy machine that just got remediated
+		return a.addPoweredOffForRemediationAnnotation(ctx, machine)
+	}
+
+	// here we know that host has been powered off and node has been deleted
+	if hasPowerOffRequestAnnotation(baremetalhost) {
+		// we can now power host back on
 		log.Printf("Requesting Host %s power on for Machine %s",
 			baremetalhost.Name, machine.Name)
-		return a.requestPowerOn(ctx, baremetalhost)
+		return a.requestPowerOn(ctx, machine, baremetalhost)
 	}
 
+	node, err := a.getNodeByMachine(ctx, machine)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
+			return err
+		}
+	}
 	//node is still not running, so we requeue
 	if node == nil {
+		if canReprovision(machine, baremetalhost) {
+			mhc := a.getMhcByMachine(machine)
```

Comment:
This looks realllly expensive. Put the expiry time rather than the request time in the powerOnRequestTimestamp annotation so we only have to do it once.

Reply:
fixed